shell/common/shell.h

```diff
@@ -169,6 +183,27 @@ class Shell final : public PlatformView::Delegate,
       CreateCallback<PlatformView> on_create_platform_view,
       CreateCallback<Rasterizer> on_create_rasterizer);
 
+  //----------------------------------------------------------------------------
+  /// @brief      Creates a shell instance using the provided settings.
+  /// @details    This version of Create can take in a running DartVMRef and a
+  ///             function that defines how the Shell's Engine should be
+  ///             created.
+  /// @param[in]  vm             A running DartVMRef where this Shell's Dart
+  ///                            code will be executed.
+  /// @param[in]  engine_maker   A function that creates an Engine during
+  ///                            initialization.
+  /// @see        Shell::Create
+  ///
+  static std::unique_ptr<Shell> Create(
+      TaskRunners task_runners,
+      const PlatformData platform_data,
+      Settings settings,
+      fml::RefPtr<const DartSnapshot> isolate_snapshot,
+      const CreateCallback<PlatformView>& on_create_platform_view,
+      const CreateCallback<Rasterizer>& on_create_rasterizer,
+      DartVMRef vm,
+      const EngineMaker& engine_maker);
```

Comment:
Same here. Could this be something like on_create_engine for consistency?

Reply:
good idea, done